### ISSUE TITLE
Layout for <input type="range">

### ISSUE THREAD

**Issue by ferjm** [servo/servo#22728]
Right now `<input type="range">` renders like a text input, while it should render like a slider with a thumb.

Servo:
<img width="160" alt="screenshot 2019-01-18 at 15 56 38" src="https://user-images.githubusercontent.com/480202/51394354-7bc3e680-1b3a-11e9-9cce-6fd687a0edb0.png">

Firefox:
![screenshot 2019-01-18 at 15 56 02](https://user-images.githubusercontent.com/480202/51394362-7ebed700-1b3a-11e9-873d-0f4c0d2dfb70.png)

**Comment by Glomels** [comment]
Is this still pending? I'm looking for something to start contributing to the project and this caught my eye.

**Comment by jdm** [comment]
Yes, it is. Unfortunately, layout code is mostly self-serve - the people still involved in this project won't really be able to help answer questions about it.

**Comment by jdm** [comment]
I think we can now implement this using the Shadow DOM similar to https://github.com/servo/servo/pull/35531 .
